lms/models/group_info.py

```diff
@@ -78,3 +80,36 @@ class GroupInfo(BASE):  # pylint:disable=too-few-public-methods
 
     #: The value of the custom_canvas_course_id param this group was last launched with.
     custom_canvas_course_id = sa.Column(sa.UnicodeText())
+
+    #: A dict of info about this group.
+    info = sa.Column(MutableDict.as_mutable(JSONB))
```

Comment:
I tried but wasn't able to write a unit test that fails without the `MutableDict` thing here. But without it (with just `sa.Column(MutableDict.as_mutable(JSONB))`) I found that, if `GroupInfo.info` already contained one instructor, then if a second instructor launched an assignment they would _not_ be added in to `GroupInfo.info`. The Python would _would_ add them to the instructors list, but it wouldn't get saved to the DB.

This is a hazard with using mutable types like JSON with SQLAlchemy. See **Detecting Changes in JSON columns when using the ORM** on this page: <https://docs.sqlalchemy.org/en/13/core/type_basics.html>. We've had to do the same fix in a couple of places in h, see: https://github.com/hypothesis/h/pull/3417